expression/expr_to_pb.go

```diff
@@ -99,18 +99,30 @@ func (pc PbConverter) ExprToPB(expr Expression) *tipb.Expr {
 	return nil
 }
 
-func (pc PbConverter) constantToPBExpr(con *Constant) *tipb.Expr {
-	var (
-		tp  tipb.ExprType
-		val []byte
-		ft  = con.GetType()
-	)
-	d, err := con.Eval(nil)
+func (pc PbConverter) conOrCorColToPBExpr(expr Expression) *tipb.Expr {
+	ft := expr.GetType()
+	d, err := expr.Eval(nil)
 	if err != nil {
 		log.Errorf("Fail to eval constant, err: %s", err.Error())
 		return nil
 	}
+	tp, val, ok := pc.encodeDatum(d)
+
+	if !ok {
+		return nil
+	}
 
+	if !pc.client.IsRequestTypeSupported(kv.ReqTypeSelect, int64(tp)) {
+		return nil
+	}
+	return &tipb.Expr{Tp: tp, Val: val, FieldType: toPBFieldType(ft)}
+}
+
+func (pc *PbConverter) encodeDatum(d types.Datum) (tipb.ExprType, []byte, bool) {
```

Comment:
can we just return `(tp tipb.ExprType, val []byte)` and check `ok` by `ok := len(val) > 0`

Reply:
No, e.g. null value just return an empty slice.

Reply:
`ok := val != nil` is another way. By the way, I think no value can be encoded to an empty slice. @coocood PTAL

Reply:
There is a `types.KindNull` which returns an empty slice ... never mind.

Reply:
Okay, for string, there's chance to return nil, so this boolean value is needed.